src/subdataset.jl

```diff
@@ -0,0 +1,41 @@
+"""
+    SubDataset{D, T, <:AbstractDataset{D,T}, <:SubArray{<:SVector{D,T},1}} <: AbstractDataset{D,T}
+
+A view of an `AbstractDataset`, as returned by the `view` function
+or the `@view` macro. A `SubDataset` is an `AbstractDataset`
+of the same type as its parent, so indexing and most functions
+can be expected to work in the same way for both the parent and the view.
+"""
+struct SubDataset{D, T, P<:AbstractDataset{D,T}, S<:SubArray{<:SVector{D,T},1}} <: AbstractDataset{D,T}
+    parent::P
```

Comment:
What is the benefit of having a parent? For the code here I don't see it being used anywhere. If it is useful then okay, but if it is not then we should remove it since it increases compile time.

Reply:
The benefit is consistency with the API of views for other types, which allow to retrieve the parent array (or data frame, etc.) with the function `parent`.

I have not used it myself so far, but I can imagine some use case where it might be useful. For instance, a function that takes a view of a dataset, does something with it, and returns another view into the same dataset "moving the window" by some stride, to help some iterative workflow. To do that you do need the parent dataset (plus the indices of the view, which are available with `parentindices`).

(That particular example might be worked around by using the parent of the `SubArray` contained in the `data` field. But it feels hackish, and the parent obtained that way would be a vector of `SVector`, not an `AbstractDataset`.)

Reply:
I'm convinced !